joinservice/internal/kubernetes/kubernetes.go

```diff
@@ -159,3 +161,37 @@ func k8sCompliantHostname(in string) (string, error) {
 	}
 	return hostname, nil
 }
+
+// CreateConfigMap creates a configmap in the kube-system namespace with the provided name and data.
+func (c *Client) CreateConfigMap(ctx context.Context, name string, data map[string]string) error {
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: constants.ConstellationNamespace,
+		},
+		Data: data,
+	}
+	_, err := c.client.CoreV1().ConfigMaps(constants.ConstellationNamespace).Create(ctx, cm, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create configmap: %w", err)
+	}
+	return nil
+}
+
+// UpdateConfigMap updates the configmap with the provided name by writing the provided key and value.
+func (c *Client) UpdateConfigMap(ctx context.Context, name, key, value string) error {
+	cm, err := c.client.CoreV1().ConfigMaps(constants.ConstellationNamespace).Get(ctx, name, metav1.GetOptions{})
```

Comment:
Depending on how frequently this resource is updated by other components, we might want to wrap this in a retry like in https://github.com/edgelesssys/constellation/blob/feat/joinservice/manage-ask/internal/kubernetes/kubectl/kubectl.go#L142.

Reply:
Yes, we should do that in the refactoring of the joinservice by directly using the kubectl utility.